roles/confluent.common/tasks/main.yml

```diff
@@ -31,6 +31,42 @@
     state: absent
     regexp: '^Defaults\s+requiretty.*$'
 
+# Process Archive File deployments
+- name: Create Confluent Platform install directory
+  file:
+    path: "{{archive_destination_path}}"
+    state: directory
+    mode: 0755
+  when: archive_file_enabled|bool
+
+# This is needed to tell the future copy task to run or not because it will fail if this file doesn't exist
+- name: Test for local Confluent Platform archive
+  stat:
+    path: "{{ confluent_archive_file_path }}"
+  register: local_confluent_archive
+
+# If the local archive exists and the target directory (i.e. creates) doesn't exist then expand the local into target
+- name: Expand local Confluent Platform archive
+  unarchive:
+    src: "{{ confluent_archive_file_path }}"
+    remote_src: yes
+    dest: "{{archive_destination_path}}"
+    mode: 0755
+    creates: "{{archive_destination_path}}/confluent-{{confluent_archive_version}}"
+  when:
+    - archive_file_enabled|bool
+    - local_confluent_archive.stat.exists|bool
+
+# If the target directory (i.e. creates) doesn't exist then download and expand the remote archive into target
+- name: Expand remote Confluent Platform archive
+  unarchive:
+    src: "{{ confluent_common_repository_archive_url }}"
+    remote_src: yes
+    dest: "{{archive_destination_path}}"
+    mode: 0755
+    creates: "{{archive_destination_path}}/confluent-{{confluent_archive_version}}"
+  when: archive_file_enabled|bool
```

Comment:
So bc this seems like duplicating code to me... I think it can be simplified down to this:

```suggestion
- name: Unarchive Confluent Platform Archive
  unarchive:
    src: "{{ confluent_common_repository_archive }}"
    remote_src: yes
    dest: "{{archive_destination_path}}"
    mode: 0755
    creates: "{{archive_destination_path}}/confluent-{{confluent_archive_version}}"
  when: archive_file_enabled|bool
```

And you can define that confluent_common_repository_archive variable as

```
### Either path to locally accessible tar file OR URL
confluent_common_repository_archive:  "{{confluent_common_repository_archive_baseurl}}/{{confluent_archive_file_name}}"
```

And we default it to the URL. 

I suppose the UX is worse because your customers will have to know what the ansible-pull path ends up being. But theyll already have to figure out what to put in their inventory file so I don't think its too bad. You are welcome to include a path example in the annotation

What are your thoughts?

Reply:
That could work. I'll make the update and push it later tonight.

Reply:
Fixed in commit 3f9b3fcd